x-pack/plugins/security/public/components/form_flyout.tsx

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { useEffect, FunctionComponent, RefObject } from 'react';
+import { FormattedMessage } from '@kbn/i18n/react';
+import {
+  EuiTitle,
+  EuiFlyout,
+  EuiFlyoutProps,
+  EuiFlyoutHeader,
+  EuiFlyoutBody,
+  EuiFlyoutFooter,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiButton,
+  EuiButtonProps,
+  EuiButtonEmpty,
+  EuiPortal,
+} from '@elastic/eui';
+
+export interface FormFlyoutProps extends Omit<EuiFlyoutProps, 'onSubmit' | 'onClose'> {
+  title: string;
+  initialFocus?: RefObject<HTMLElement>;
+  onCancel(): void;
+  onSubmit(): void;
+  submitButtonText: string;
+  submitButtonColor?: EuiButtonProps['color'];
+  isLoading?: EuiButtonProps['isLoading'];
+  isDisabled?: EuiButtonProps['isDisabled'];
+}
+
+export const FormFlyout: FunctionComponent<FormFlyoutProps> = ({
+  title,
+  submitButtonText,
+  submitButtonColor,
+  onCancel,
+  onSubmit,
+  isLoading,
+  isDisabled,
+  children,
+  initialFocus,
+  ownFocus = true,
+  ...rest
+}) => {
+  useEffect(() => {
+    if (initialFocus && initialFocus.current) {
+      initialFocus.current.focus();
+    }
+  }, [initialFocus]);
+
+  const flyout = (
+    <EuiFlyout ownFocus={ownFocus} onClose={onCancel} {...rest}>
+      <EuiFlyoutHeader hasBorder>
+        <EuiTitle size="m">
+          <h2>{title}</h2>
```

Comment:
For accessibility we recommend adding id's and aria-attributes like so:

```suggestion
    <EuiFlyout ownFocus={ownFocus} onClose={onCancel} aria-labelledby="flyoutTitle" {...rest}>
      <EuiFlyoutHeader hasBorder>
        <EuiTitle size="m">
          <h2 id="flyoutTitle">{title}</h2>
```

Though of course you'll want to allow passing in a custom id or generating one per instance.

Reply:
Awesome, thanks, I'll add that in!